src/tv2-common/__tests__/EvaluateCueTelemetrics.spec.ts

```diff
@@ -0,0 +1,262 @@
+import { EvaluateCueTelemetrics } from '../cues/EvaluateCueTelemetrics'
+import { CueDefinitionTelemetrics } from '../inewsConversion'
+
+describe('EvaluateCueTelemetrics', () => {
+	let context: IShowStyleUserContext
+
+	beforeEach(() => {
+		context = ({} as unknown) as IShowStyleUserContext
+	})
+
+	it('adds a Telemetrics piece', () => {
+		const cueDefinition: CueDefinitionTelemetrics = createTelemetricsCueDefinition()
+		const pieces: IBlueprintPiece[] = []
+
+		EvaluateCueTelemetrics(context, cueDefinition, pieces, '')
+
+		expect(pieces).toHaveLength(1)
+	})
+
+	function createTelemetricsCueDefinition(presetShot?: number, startTimeInSeconds?: number): CueDefinitionTelemetrics {
+		return {
+			type: CueType.Telemetrics,
+			presetIdentifier: presetShot ?? 1,
+			iNewsCommand: '',
+			start: {
+				seconds: startTimeInSeconds
+			}
+		}
+	}
+
+	it('parse the externalId to the piece', () => {
+		const cueDefinition: CueDefinitionTelemetrics = createTelemetricsCueDefinition()
+		const externalId: string = 'someExternalId'
+		const pieces: IBlueprintPiece[] = []
+
+		EvaluateCueTelemetrics(context, cueDefinition, pieces, externalId)
+
+		expect(pieces[0].externalId).toEqual(externalId)
+	})
+
+	it('has lifeSpan withinPart', () => {
+		const cueDefinition: CueDefinitionTelemetrics = createTelemetricsCueDefinition()
+		const pieces: IBlueprintPiece[] = []
+
+		EvaluateCueTelemetrics(context, cueDefinition, pieces, '')
+
+		expect(pieces[0].lifespan).toEqual(PieceLifespan.WithinPart)
+	})
+
+	it('creates a telemetrics timeline object', () => {
+		const cueDefinition: CueDefinitionTelemetrics = createTelemetricsCueDefinition()
+		const pieces: IBlueprintPiece[] = []
+
+		EvaluateCueTelemetrics(context, cueDefinition, pieces, '')
+
+		expect(pieces[0].content.timelineObjects).toHaveLength(1)
+		const result = pieces[0].content.timelineObjects[0]
+		expect(result.layer).toEqual(RobotCameraLayer.TELEMETRICS)
+		expect(result.content.deviceType).toEqual(DeviceType.TELEMETRICS)
+	})
+
+	it('timeline object has start time 0', () => {
+		const cueDefinition: CueDefinitionTelemetrics = createTelemetricsCueDefinition()
+		const pieces: IBlueprintPiece[] = []
+
+		EvaluateCueTelemetrics(context, cueDefinition, pieces, '')
+
+		expect((pieces[0].content.timelineObjects[0].enable as TimelineEnable).start).toEqual(0)
+	})
+
+	it('has Telemetrics source layer', () => {
+		const cueDefinition: CueDefinitionTelemetrics = createTelemetricsCueDefinition()
+		const pieces: IBlueprintPiece[] = []
+
+		EvaluateCueTelemetrics(context, cueDefinition, pieces, '')
+
+		expect(pieces[0].sourceLayerId).toEqual(AFVDSourceLayer.Telemetrics)
+	})
+
+	it('has SEC for output layer', () => {
+		const cueDefinition: CueDefinitionTelemetrics = createTelemetricsCueDefinition()
+		const pieces: IBlueprintPiece[] = []
+
+		EvaluateCueTelemetrics(context, cueDefinition, pieces, '')
+
+		expect(pieces[0].outputLayerId).toEqual(SharedOutputLayers.SEC)
+	})
```

Comment:
From a maintenance perspective, I think it comes down to personal preference. I would personally prefer to maintain this, than one method that asserts all the attributes.

In your suggestion we would have something like this:
```
const expectedResult: IBlueprintPiece = {
			externalId,
			name: 'Robot[1]',
			lifespan: PieceLifespan.WithinPart,
			sourceLayerId: AFVDSourceLayer.Telemetrics,
			outputLayerId: SharedOutputLayers.SEC,
			enable: {
				start: 0
			},
			content: {
				timelineObjects: [
					{
						id: 'telemetrics_preset_1',
						layer: RobotCameraLayer.TELEMETRICS,
						content: {
							deviceType: DeviceType.TELEMETRICS
						},
						enable: {
							start: 0
						}
					}
				]
			}
		}
expect(result).toMatchObject(expectedResult)
```
My issue with this is that it's only the non-dynamic attributes that would be tested enough with this test. The dynamic attributes (name, enable.start and  timelineObject.id) would all still need more tests to assert that given different inputs, it produce the correct output.
So we have test that covers some of the attributes, but not all. How would a maintenance developer know which attributes it covers and which it doesn't? 

Also values like `Robot[1]` seems completely random and out of the blue, but the test fails if we change it. But if the test is just focused around the name, then we can give the test a name like `receives cameraPreset 1, blueprint piece name is Robot[1]` and the value makes a lot of sense all of a sudden.

Another issue I have with this is the name of the test. What is a suitable name for a test that test some attributes, but not all? My best bet is something like `Assert default values`. But a name like that is very non-descriptive. 
Say we use a third party tool to run our test in a CI pipeline (i.e. GitHub Actions) and you get a report containing the names of the failed tests. If you just receive `The test 'EvaluateCueTelemetrics.assert default values' failed` you would have to hope it also included the test output or be forced to go look at the implementation of the test and guess which one of the many attributes failed.
If you instead get a message like `The test 'EvaluateCueTelemetrics.parse the externalId to the piece' failed` you then have a pretty good idea of what is wrong, without looking at anything else.

As a maintenance developer I would also be more hesitant to change a test like this, since there are many more 'parameters' that can make the test fail.

But as stated in the beginning I think it comes down to personal preference. Maybe we should have a meeting and make a team agreement?